apps/xmtp.chat/src/components/Messages/WalletSendCalls.tsx

```diff
@@ -0,0 +1,57 @@
+import { Button } from "@mantine/core";
+import type { Conversation } from "@xmtp/browser-sdk";
+import {
+  ContentTypeTransactionReference,
+  type TransactionReference,
+} from "@xmtp/content-type-transaction-reference";
+import type { WalletSendCallsParams } from "@xmtp/content-type-wallet-send-calls";
+import { useSendTransaction } from "wagmi";
+
+export type WalletSendCallsProps = {
+  content: WalletSendCallsParams;
+  sendMessage: Conversation["send"];
+};
+
+export const WalletSendCallsUI: React.FC<WalletSendCallsProps> = ({
+  content,
+  sendMessage,
+}) => {
+  const { sendTransactionAsync } = useSendTransaction();
+
+  function handleSubmit() {
```

Comment:
can you set this up as:

```
const handleSubmit = useCallback(async () => {...}, [...]);
```

this follows convention with other `async` callbacks in this app.

Reply:
Tried the above syntax but that triggers a eslint error about no-misused-promises. `Promise-returning function provided to attribute where a void return was expected.`. Updated the code to use `useCallback` and using [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) pattern inside. Please check if this is fine.

Reply:
in order for ESLint to be happy, you need to update the the `onClick` handler to this:

```
<Button
  onClick={() => {
    void handleSubmit();
  }}>
  Submit
</Button>
```